src/main/java/com/github/angleshq/angles/util/AnglesReporterUtils.java

```diff
@@ -1,24 +1,32 @@
 package com.github.angleshq.angles.util;
 
 import com.github.angleshq.angles.AnglesReporter;
+import com.github.angleshq.angles.AnglesReporterInterface;
 import com.github.angleshq.angles.exceptions.AnglesPropertyNotGivenException;
 
 import static org.apache.commons.lang3.StringUtils.isBlank;
 
 public class AnglesReporterUtils {
 
-    private static AnglesReporter anglesReporter;
+    private static AnglesReporterInterface anglesReporter;
     private static String runName;
     private static String team;
     private static String component;
     private static String environment;
 
-    public static AnglesReporter initialiseAnglesTestParameters() throws AnglesPropertyNotGivenException {
-        anglesReporter = AnglesReporter.getInstance(getAnglesPropertyFromSystem("angles.url") + "/rest/api/v1.0/");
-        runName = getAnglesPropertyFromSystem("angles.runName");
-        team = getAnglesPropertyFromSystem("angles.team");
-        component = getAnglesPropertyFromSystem("angles.component");
-        environment = getAnglesPropertyFromSystem("angles.environment");
+    public static AnglesReporterInterface initialiseAnglesTestParameters() throws AnglesPropertyNotGivenException {
+        String enabled = getAnglesPropertyFromSystem("angles.enabled");
```

Comment:
As discussed: this throws an AnglesPropertyNotGivenException in case the angles.enabled argument is missing ( which will bypass the creation of the empty reporter and lead to a null pointer exception).

Reply:
fixed.